services/navigation/builtin/builtin.go

```diff
@@ -581,5 +642,90 @@ func (svc *builtIn) Obstacles(ctx context.Context, extra map[string]interface{})
 func (svc *builtIn) Paths(ctx context.Context, extra map[string]interface{}) ([]*navigation.Path, error) {
 	svc.mu.RLock()
 	defer svc.mu.RUnlock()
-	return nil, errUnimplemented
+
+	rawExecutionWaypoint := svc.activeExecutionWaypoint.Load()
+	// If there is no execution, return empty paths
+	if rawExecutionWaypoint == nil || rawExecutionWaypoint == emptyExecutionWaypoint {
```

Comment:
I don't think we need the second check here we should be able to control what is put into an executionWaypoint and this should never come up

Reply:
The invariant we want to be maintained is that nav only has one Execution in progress at a time & never loses track of an execution it started.

In order for that invariant to be maintained, we need to ensure that we are only setting `activeExecutionWaypoint` when it is nil / empty & only unsetting it (setting it to empty) when it is set to the executionID we expected.

This checks those invariants & gives us feedback if someone in the future changes the code such that those invariants are not maintained.

 I can remove these checks as the current form of the code maintains those invariants & never goes down the sad paths if you would like me to, however, I would prefer keeping these checks so that a future developer that changes this code will get feedback if they change the code to violate the invariants.

Reply:
Ok that seems reasonable thanks for the exaplanation